src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java

```diff
@@ -0,0 +1,187 @@
+    private static final String ACH_EXTRACT_TEST_PATH_AND_FILE_FOR_VALIDATION = ACH_EXTRACT_TEST_PATH_AND_FILE + ".READY";
+    private static final SimpleDateFormat yyyyMMddHHmmss_DATE_FORMATTER = new SimpleDateFormat("yyyyMMddHHmmss");
+    private static final SimpleDateFormat yyyyMMdd_DATE_FORMATTER = new SimpleDateFormat("yyyyMMdd");
+    private static final String[] NOTIFICATION_EMAIL_ADDRESSES = {"abc@xyz.edu", "def@xyz.edu", "ghi@xyz.edu"};
+    private static final int HEADER_LINE_IN_OUTPUT_FILE = 1;
+    private static final int FIRST_PAYEE_LINE_IN_OUTPUT_FILE = HEADER_LINE_IN_OUTPUT_FILE + NOTIFICATION_EMAIL_ADDRESSES.length + 1;
+    private static final  Logger LOG = org.apache.log4j.Logger.getLogger(CuExtractPaymentServiceImplTest.class);
+    
+    
+    
+    private CuExtractPaymentServiceImpl cuExtractPaymentServiceImpl;
+
+    @Before
+    public void setUp() throws Exception {
+        cuExtractPaymentServiceImpl = new TestCuExtractPaymentServiceImpl();
+        File testOutputDirectory = new File(ACH_EXTRACT_TEST_FILE_PATH);
+        testOutputDirectory.mkdir();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cuExtractPaymentServiceImpl = null;
+        FileUtils.forceDelete(new File(ACH_EXTRACT_TEST_FILE_PATH).getAbsoluteFile());
+    }
+
+    @Test
+    public void testCalculateHeaderDate() {
+        Date now = new Date(Calendar.getInstance().getTimeInMillis());
+        Date functionReturnedDate = cuExtractPaymentServiceImpl.calculateHeaderDate(now);
+        Calendar calendarCalculator = Calendar.getInstance();
+        calendarCalculator.setTimeInMillis(now.getTime());
+        calendarCalculator.add(Calendar.DATE, 1);
+        Date expectedDate = new Date(calendarCalculator.getTimeInMillis());
+        assertEquals("headerDate from function should equal date we calculate in the same manner:  ", expectedDate, functionReturnedDate);
+    }
+    
+    @Test
+    public void testAchFileHeaderDateIsOneDayAfterProcessDate() {
+        cuExtractPaymentServiceImpl.setAchBundlerHelperService(new TestAchBundlerHelperService());
+        
+        PaymentStatus extractedStatus = new PaymentStatus();
+        Date processDate = new Date(Calendar.getInstance().getTimeInMillis());
+        List<String> notificationEmailAddresses = new ArrayList<String>(Arrays.asList(NOTIFICATION_EMAIL_ADDRESSES));
+        
+        cuExtractPaymentServiceImpl.writeExtractAchFileMellonBankFastTrack(extractedStatus, ACH_EXTRACT_TEST_PATH_AND_FILE, processDate, yyyyMMddHHmmss_DATE_FORMATTER, notificationEmailAddresses);
+        
+        Date expectedHeaderDate = cuExtractPaymentServiceImpl.calculateHeaderDate(processDate);
+        assertEquals("ACH Extract Payments File Header Date is not what was expected..", true, isDateInOutputFileWhereExpected(HEADER_LINE_IN_OUTPUT_FILE, yyyyMMddHHmmss_DATE_FORMATTER.format(expectedHeaderDate)));
```

Comment:
This is a cool unit test, I like it.  I am confused about this line though  Do we expect the "expectedHeaderDate" to be in the header row of the output file?

Not worth changing (IMHO) unless other code review things come up, but I would use assertTrue instead of assertEquals here.